bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/discovery/MiIoDiscovery.java

```diff
@@ -66,11 +76,33 @@
 
     private final Logger logger = LoggerFactory.getLogger(MiIoDiscovery.class);
     private final CloudConnector cloudConnector;
+    private Map<String, String> cloudDevices = new HashMap<>();
```

Comment:
If there is change this would be modified by multiple threads the you should use ConcurrentHashMap instead.